hazelcast/serialization/portable/writer.py

```diff
@@ -145,6 +194,23 @@ def write_utf(self, field_name, value):
     def write_utf_array(self, field_name, values):
         self.write_string_array(field_name, values)
 
+    def write_nullable_field(self, field_name: str, field_type: FieldType, value, func):
+        self._set_position(field_name, field_type)
+        is_nil = value is None
+        self._out.write_boolean(is_nil)
+        if not is_nil:
+            func(self._out, value)
+
+    def write_object_array_field(self, field_name, field_type, values, func):
+        self._set_position(field_name, field_type)
+        length = NULL_ARRAY_LENGTH if values is None else len(values)
+        if length < 0:
+            self._out.write_int(NULL_ARRAY_LENGTH)
+            return
+        self._out.write_int(length)
```

Comment:
maybe always write the length, and if it is < 0, return

Reply:
Fixed at https://github.com/hazelcast/hazelcast-python-client/commit/6b34418b0dd46c23cb7a5ed87e84d83865ce27e1.